webapp/src/components/confirm_modal.tsx

```diff
@@ -0,0 +1,202 @@
+
+    /*
+     * Text/jsx element on the confirm button
+     */
+    confirmButtonText: React.ReactNode;
+
+    /*
+     * Text/jsx element on the cancel button
+     */
+    cancelButtonText: React.ReactNode;
+
+    /*
+     * Set to show checkbox
+     */
+    showCheckbox?: boolean;
+
+    /*
+     * Text/jsx element to display with the checkbox
+     */
+    checkboxText?: string;
+
+    /*
+     * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
+     */
+    onConfirm?: (arg0: boolean) => void;
+
+    /*
+     * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
+     */
+    onCancel?: (arg0: boolean) => void;
+
+    /**
+     * Function called when modal is dismissed
+     */
+    onExited?: () => void;
+
+    /*
+     * Set to hide the cancel button
+     */
+    hideCancel: boolean;
+
+}
+
+export type State = {
+}
+
+export default class ConfirmModal extends PureComponent<Props, State> {
+    componentDidMount() {
+        if (this.props.show) {
+            document.addEventListener('keydown', this.handleKeypress);
+        }
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeypress);
+    }
+
+    shouldComponentUpdate(nextProps) {
+        return nextProps.show !== this.props.show;
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
+        if (this.props.show && !nextProps.show) {
+            document.removeEventListener('keydown', this.handleKeypress);
+        } else if (!this.props.show && nextProps.show) {
+            document.addEventListener('keydown', this.handleKeypress);
+        }
+    }
+
+    handleKeypress = (e) => {
+        if (e.key === 'Enter' && this.props.show) {
+            this.handleConfirm();
+        }
+    }
+
+    handleConfirm = () => {
+        // const checked = this.refs.checkbox ? this.refs.checkbox.checked : false;
+        const checked = false;
+        this.props.onConfirm(checked);
+    }
+
+    handleCancel = () => {
+        // const checked = this.refs.checkbox ? this.refs.checkbox.checked : false;
+        const checked = false;
+        this.props.onCancel(checked);
+    }
+
+    render() {
+        let checkbox;
+        if (this.props.showCheckbox) {
+            checkbox = (
+                <div className='checkbox text-right margin-bottom--none'>
+                    <label>
+                        <input
+                            ref='checkbox'
+                            type='checkbox'
+                        />
+                        {this.props.checkboxText}
+                    </label>
+                </div>
+            );
+        }
+
+        let cancelText;
+        if (this.props.cancelButtonText) {
+            cancelText = this.props.cancelButtonText;
+        } else {
+            cancelText = (
+                <FormattedMessage
```

Comment:
Nice job taking a step to i18n the webapp. I think this could be a good help wanted campaign once we have the infrastructure set up for one/several repo(s).

I wonder if we get the webapp's `i18n` configs for free without any additions due to the way i18n is hooked into the component tree. @levb I think that is worth exploring for reuse of common translations.

Reply:
Let's have a separate conversation about this, we have https://github.com/mattermost/mattermost-plugin-jira/issues/317 for it, @crspeller and @hanzei need to work out their approach to the server-side localization.